include/openssl/kdf.h

```diff
@@ -90,6 +93,19 @@ extern "C" {
             EVP_PKEY_CTX_ctrl_uint64(pctx, -1, EVP_PKEY_OP_DERIVE, \
                             EVP_PKEY_CTRL_SCRYPT_MAXMEM_BYTES, maxmem_bytes)
 
+# define EVP_PKEY_CTX_set_pbkdf2_md(pctx, md) \
+            EVP_PKEY_CTX_ctrl(pctx, -1, EVP_PKEY_OP_DERIVE, \
+                              EVP_PKEY_CTRL_PBKDF2_MD, 0, (void *)(md))
+
+# define EVP_PKEY_CTX_set1_pbkdf2_salt(pctx, salt, saltlen) \
```

Comment:
Hmmm -- but what about HKDF? HKDF already uses EVP_PKEY_CTRL_HKDF_SALT (EVP_PKEY_ALG_CTRL + 4).

In order to be consistent, shouldn't that then also use the generic CTRL_SALT? Because then we have a conflict: currently on master, both EVP_PKEY_CTRL_HKDF_SALT (+4) and EVP_PKEY_CTRL_SCRYPT_SALT (+9) are used. So consolidating the two might break something either way (although scrypt has only been in the code for a very short number of days, so not that dramatic probably). If you have an idea how you'd want this to look, I'd get cracking.